### ISSUE TITLE
Eliminate distinction between Number and Integer?

### ISSUE THREAD

**Issue by ronichoudhury** [Kitware/candela#277]
"Integer" is sort of a false distinction in JavaScript, since integers are represented by `Number`s. Eliminating the distinction in Resonant Lab would have two benefits:
- The UI would become less cluttered and strictly easier to understand
- The code would become a little bit shorter

**Comment by alex-r-bigelow** [comment]
For the record, we were using "integer" distinct from "number" because we wanted our types to conform to those in use in another project (I think it was Minerva?), but I can't seem to find the documentation page that listed the types.

**Comment by manthey** [comment]
Are there any visualizations that treat integers as distinct from numbers
or only handle integers?

On Tue, Jun 28, 2016 at 5:05 PM, Alex Bigelow <notifications@github.com>
wrote:

> For the record, we were using "integer" distinct from "number" because we
> wanted our types to conform to those in use in another project (I think it
> was Minerva?), but I can't seem to find the documentation page that listed
> the types.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/candela/issues/277#issuecomment-229183331>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIX_R9cIk5T5kRhVKJf5volocwp6uY92ks5qQYyggaJpZM4JAhtq>
> .
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

**Comment by ronichoudhury** [comment]
I don't think any of the ones [in Candela](https://github.com/Kitware/candela/tree/master/src/candela/components) do, aside from treating an integer column as qualitative (for, e.g., the bar chart to identify distinct bars). But that operation is general with respect to type so doesn't require drawing the Number/Integer distinction.

**Comment by alex-r-bigelow** [comment]
Yeah, I was thinking the same thing—it's possible a visualization might care about it for qualitative vs quantitative reasons, but if you're trying to use floats as qualitative values, you probably have bigger problems than just type coercion (this is where we'd punt to Resonant Data Reshaper or something).

I think UpSet allows integers, but not numbers, for set identification, but really we're just using integers there as proxies for boolean values (the "correct" way is probably to auto-detect/specify that those values are actually booleans. Type auto-detection currently isn't that smart, but with the histograms available, it might not actually be that hard).

The only other use for it that I can imagine is where you're expecting the data to all be integers, but you actually have a few floats in there. Resonant Lab would probably auto-detect the field as a number instead of an integer, and the histogram would be able to explain why. Without the distinction, you'd have no way of knowing about the floats until you tried to visualize the page of data that contains the weird values.

With regard to clutter / understanding difficulty, I don't see much of a difference—we're already making people differentiate between numbers, booleans, strings, etc... it's just one more category. As all the integer stuff is already implemented (unless there's more work to do that I'm unaware of), my vote would be to leave it in there, and make sure we ask about it in the next round of user testing.

**Comment by manthey** [comment]
I've already found Resonant Lab useful for detecting bad data in a data
set.  If I'm expecting integers and the column appears as number, that is
quite useful to me, even if it has no other utility.

On Fri, Jul 1, 2016 at 10:43 AM, Alex Bigelow <notifications@github.com>
wrote:

> Yeah, I was thinking the same thing—it's possible a visualization might
> care about it for qualitative vs quantitative reasons, but if you're trying
> to use floats as qualitative values, you probably have bigger problems than
> just type coercion (this is where we'd punt to Resonant Data Reshaper or
> something).
>
> I think UpSet allows integers, but not numbers, for set identification,
> but really we're just using integers there as proxies for boolean values
> (the "correct" way is probably to auto-detect/specify that those values are
> actually booleans. Type auto-detection currently isn't that smart, but with
> the histograms available, it might not actually be that hard).
>
> The only other use for it that I can imagine is where you're expecting the
> data to all be integers, but you actually have a few floats in there.
> Resonant Lab would probably auto-detect the field as a number instead of an
> integer, and the histogram would be able to explain why. Without the
> distinction, you'd have no way of knowing about the floats until you tried
> to visualize the page of data that contains the weird values.
>
> With regard to clutter / understanding difficulty, I don't see much of a
> difference—we're already making people differentiate between numbers,
> booleans, strings, etc... it's just one more category. As all the integer
> stuff is already implemented (unless there's more work to do that I'm
> unaware of), my vote would be to leave it in there, and make sure we ask
> about it in the next round of user testing.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/candela/issues/277#issuecomment-229965122>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIX_R8cUiA7PpoWYu8A0J_GamBckyLHfks5qRSd0gaJpZM4JAhtq>
> .
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

**Comment by ronichoudhury** [comment]
Since we have at least one vote for its usefulness, let's leave it in for
now. My concern about clutter isn't the absolute number of icons, it's the
visual similarity of the icons combined with (my perceived) lack of "power"
in the distinction between the associated types. But let's definitely
ask/observe about it in the next round of user testing.

Let's leave the issue open until then, since it feels like that round of
user testing will be the final answer on this.

**Comment by alex-r-bigelow** [comment]
A new icon's always an easy fix—I'm open to suggestions.

**Comment by ronichoudhury** [comment]
The \mathbb{R} and \mathbb{Z} glyphs are one idea, though that may be even
more obscure than having to learn the current ones.
